src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -0,0 +1,132 @@
+    {
+        ::ITextRangeProvider* returnVal;
+        // ScreenInfoUiaProvider doesn't actually use parameter, so just pass in nullptr
+        _uiaProvider.RangeFromChild(/* IRawElementProviderSimple */ nullptr,
+                                    &returnVal);
+
+        auto parentProvider = this->ProviderFromPeer(*this);
+        auto xutr = winrt::make_self<XamlUiaTextRange>(returnVal, parentProvider);
+        return xutr.as<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider>();
+    }
+
+    Windows::UI::Xaml::Automation::Provider::ITextRangeProvider TermControlAutomationPeer::RangeFromPoint(Windows::Foundation::Point screenLocation)
+    {
+        ::ITextRangeProvider* returnVal;
+        _uiaProvider.RangeFromPoint({ screenLocation.X, screenLocation.Y }, &returnVal);
+
+        auto parentProvider = this->ProviderFromPeer(*this);
+        auto xutr = winrt::make_self<XamlUiaTextRange>(returnVal, parentProvider);
+        return xutr.as<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider>();
+    }
+
+    Windows::UI::Xaml::Automation::Provider::ITextRangeProvider TermControlAutomationPeer::DocumentRange()
+    {
+        ::ITextRangeProvider* returnVal;
+        _uiaProvider.get_DocumentRange(&returnVal);
+
+        auto parentProvider = this->ProviderFromPeer(*this);
+        auto xutr = winrt::make_self<XamlUiaTextRange>(returnVal, parentProvider);
+        return xutr.as<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider>();
+    }
+
+    Windows::UI::Xaml::Automation::SupportedTextSelection TermControlAutomationPeer::SupportedTextSelection()
+    {
+        ::SupportedTextSelection returnVal;
+        _uiaProvider.get_SupportedTextSelection(&returnVal);
+        return static_cast<Windows::UI::Xaml::Automation::SupportedTextSelection>(returnVal);
+    }
+
+#pragma endregion
+
+    RECT TermControlAutomationPeer::GetBoundingRectWrapped()
+    {
+        auto rect = GetBoundingRectangle();
+        return {
+            gsl::narrow<LONG>(rect.X),
+            gsl::narrow<LONG>(rect.Y),
+            gsl::narrow<LONG>(rect.X + rect.Width),
+            gsl::narrow<LONG>(rect.Y + rect.Height)
+        };
+    }
+
+    winrt::com_array<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider> TermControlAutomationPeer::SafeArrayToComArray(SAFEARRAY* textRanges)
+    {
+        auto owningVector = SafeArrayToOwningVector<::UiaTextRange>(textRanges);
+        int count = owningVector.size();
+
+        std::vector<Windows::UI::Xaml::Automation::Provider::ITextRangeProvider> vec;
+        vec.reserve(count);
+        auto parentProvider = this->ProviderFromPeer(*this);
+        for (int i = 0; i < count; i++)
+        {
+            auto provider = owningVector[i];
+            auto xutr = winrt::make_self<XamlUiaTextRange>(provider.get(), parentProvider);

```

Comment:
nonono wait use `provider.detach()` to _take ownership out of the vector_

yuo also need to make provider an `auto&` so you're referring to the real/actual member of the vector

Reply:
How's this? Renamed `owningVector` --> `providers` and directly called `detach()`.